internal/controller/pod_event_handler.go

```diff
@@ -0,0 +1,137 @@
+package controller
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	workspacesv1alpha1 "github.com/jupyter-ai-contrib/jupyter-k8s/api/v1alpha1"
+)
+
+// Variable for dependency injection in tests
+var (
+	newSSMRemoteAccessStrategy = func() (*SSMRemoteAccessStrategy, error) {
+		return NewSSMRemoteAccessStrategy(nil, nil)
+	}
+)
+
+// PodEventHandler handles pod events for workspace pods
+type PodEventHandler struct {
+	client                  client.Client
+	resourceManager         *ResourceManager
+	ssmRemoteAccessStrategy *SSMRemoteAccessStrategy
+}
+
+// NewPodEventHandler creates a new PodEventHandler
+func NewPodEventHandler(k8sClient client.Client, resourceManager *ResourceManager) *PodEventHandler {
+	ssmStrategy, err := newSSMRemoteAccessStrategy()
+	if err != nil {
+		logf.Log.Error(err, "Failed to initialize SSM remote access strategy - SSM features will be disabled")
+		ssmStrategy = nil
+	}
+
+	return &PodEventHandler{
+		client:                  k8sClient,
+		resourceManager:         resourceManager,
+		ssmRemoteAccessStrategy: ssmStrategy,
+	}
+}
+
+// HandleWorkspacePodEvents handles pod events for workspace pods
+func (h *PodEventHandler) HandleWorkspacePodEvents(ctx context.Context, obj client.Object) []reconcile.Request {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		return nil
+	}
+
+	logger := logf.FromContext(ctx).WithValues("pod", pod.Name, "namespace", pod.Namespace)
+	logger.V(1).Info("Received pod event", "phase", pod.Status.Phase)
+
+	// Check if pod has workspace labels (early filtering)
+	workspaceName, hasWorkspace := pod.Labels[LabelWorkspaceName]
```

Comment:
We should filter this upstream, at the `controller` level. Could you look at [predicate](https://book.kubebuilder.io/reference/watching-resources/predicates-with-watch)?

Not a blocker for this PR, but we definitely need it before GA :)